From b0481f7b2c090d6ccfea0b3c5528f7e561bcec46 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Thu, 7 Oct 2021 15:59:06 +0200
Subject: [PATCH 0646/1695] dts: stm32mp1: add the calibration node

Add the node to manage the calibration at soc level.
It defines the timer used by default with the correct
entries.
Add a label to timer for calibration reference.
Enable the associated timers by default to manage
the calibration.
Calib disabled at soc level.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 15ca212ce054786f22e9c14663f8e9f64555a2c6 dts: stm32mp1: add the calibration node]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@st.com>
Change-Id: I414a6e7dae58d06fafd3d98e7119588d7ee994ae
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/240879
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/286372
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/362465
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
---
 core/arch/arm/dts/stm32mp131.dtsi | 16 ++++++++++++++++
 core/arch/arm/dts/stm32mp151.dtsi | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp131.dtsi b/core/arch/arm/dts/stm32mp131.dtsi
index 03dacc6b4..b38df95ac 100644
--- a/core/arch/arm/dts/stm32mp131.dtsi
+++ b/core/arch/arm/dts/stm32mp131.dtsi
@@ -83,6 +83,22 @@
 
 	};
 
+	osc_calibration: osc-calibration {
+		compatible = "st,osc-calibration";
+
+		csi_calibration: csi-calibration {
+			compatible = "st,csi-cal";
+			counter = <&timers12 0 2>;
+			status = "disabled";
+		};
+
+		hsi_calibration: hsi-calibration {
+			compatible = "st,hsi-cal";
+			counter = <&timers12 0 1>;
+			status = "disabled";
+		};
+	};
+
 	scmi_regu: scmi-regu {
 		compatible = "st,scmi-regulator-consumer";
 		#address-cells = <1>;
diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index ae461cf82..6f39c0cbe 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -115,6 +115,22 @@
 		status = "disabled";
 	};
 
+	osc_calibration: osc-calibration {
+		compatible = "st,osc-calibration";
+
+		csi_calibration: csi-calibration {
+			compatible = "st,csi-cal";
+			counter = <&timers15 0 8>;
+			status = "disabled";
+		};
+
+		hsi_calibration: hsi-calibration {
+			compatible = "st,hsi-cal";
+			counter = <&timers15 0 7>;
+			status = "disabled";
+		};
+	};
+
 	soc {
 		compatible = "simple-bus";
 		#address-cells = <1>;
-- 
2.47.2

